### PR TITLE
CON-1466 training provider page changes

### PR DESCRIPTION
**Pull request by shomavg**

### CODE REVIEW

File: src/SFA.DAS.ProviderRelationships.Types/Dtos/AccountProviderDto.cs [hunk 1]
```diff
@@ -8,5 +8,6 @@ public class AccountProviderDto
         public long ProviderUkprn { get; set; }
         public string ProviderName { get; set; }
         public List<AccountLegalEntityDto> AccountLegalEntities { get; set; }
+        public string FormattedProviderSuggestion { get; set; }
```
Comment (cofaulco) [inline src/SFA.DAS.ProviderRelationships.Types/Dtos/AccountProviderDto.cs]: This should be on the ProviderDto model

File: src/SFA.DAS.ProviderRelationships.UnitTests/Application/Queries/GetProviderToAddQueryHandlerTests.cs [hunk 2]
```diff
@@ -47,7 +47,7 @@ public GetProviderToAddQueryHandlerTestsFixture()
         {
             Query = new GetProviderToAddQuery(12345678);
             Db = new ProviderRelationshipsDbContext(new DbContextOptionsBuilder<ProviderRelationshipsDbContext>().UseInMemoryDatabase(Guid.NewGuid().ToString()).ConfigureWarnings(warnings => warnings.Throw(RelationalEventId.QueryClientEvaluationWarning)).Options);
-            ConfigurationProvider = new MapperConfiguration(c => c.AddProfiles(typeof(ProviderMappings)));
```
Comment (cofaulco) [inline src/SFA.DAS.ProviderRelationships.UnitTests/Application/Queries/GetProviderToAddQueryHandlerTests.cs]: I think the ProviderMappings will still be necessary

File: src/SFA.DAS.ProviderRelationships.Web/ViewModels/AccountProviders/FindProvidersViewModel.cs [hunk 3]
```diff
@@ -1,16 +1,19 @@
...
+    public class FindProvidersViewModel : IAuthorizationContextModel
     {
         [Required]
         public long? AccountId { get; set; }
-        
+
         [Required(ErrorMessage = ErrorMessages.InvalidUkprn)]
         [RegularExpression(@"[\d+]{8}", ErrorMessage = ErrorMessages.InvalidUkprn)]
         public string Ukprn { get; set; }
+        public List<AccountProviderDto> Providers { get; set; }
```
Comment (cofaulco) [inline src/SFA.DAS.ProviderRelationships.Web/ViewModels/AccountProviders/FindProvidersViewModel.cs]: Should be ProviderDto

File: src/SFA.DAS.ProviderRelationships/Application/Queries/GetAllProviders/FindAllProvidersQuery.cs [hunk 4]
```diff
@@ -0,0 +1,12 @@
+using MediatR;
+
+namespace SFA.DAS.ProviderRelationships.Application.Queries.GetAllProviders
+{
+    public class FindAllProvidersQuery : IRequest<FindAllProvidersQueryResult>
```
[the remark this answers is not in the document]
Comment (shomavg) [reply src/SFA.DAS.ProviderRelationships/Application/Queries/GetAllProviders/FindAllProvidersQuery.cs]: 👍

File: src/SFA.DAS.ProviderRelationships.Web/ViewModels/AccountProviders/FindProvidersViewModel.cs (hunk 3 above)
Comment (shomavg) [reply src/SFA.DAS.ProviderRelationships.Web/ViewModels/AccountProviders/FindProvidersViewModel.cs]: 👍

File: src/SFA.DAS.ProviderRelationships/Application/Queries/GetAllProviders/GetAllProvidersQueryHandler.cs [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+            foreach (var provider in providers)
+            {
+                var providerDto = new AccountProviderDto {
+                    ProviderUkprn = provider.Ukprn,
+                    ProviderName = provider.Name
+                };
+                providersDto.Add(providerDto);
+            }
+            return new GetAllProvidersQueryResult(providersDto);
```
Comment (cofaulco) [inline src/SFA.DAS.ProviderRelationships/Application/Queries/GetAllProviders/GetAllProvidersQueryHandler.cs]:
```suggestion
           return GetAllProvidersQueryResult(
           _db.Value.Providers
           .Select(p => new ProviderDto { ProviderUkprn = p.Ukprn, ProviderName = p.Name })
           .ToList());
```

Comment (cofaulco) [reply src/SFA.DAS.ProviderRelationships/Application/Queries/GetAllProviders/GetAllProvidersQueryHandler.cs]: This should return a ProviderDTO. Nothing to do with an account

File: src/SFA.DAS.ProviderRelationships.Web/Views/AccountProviders/Find.cshtml [hunk 6]
```diff
@@ -16,43 +16,47 @@
     </div>
 
     <div class="govuk-grid-column-full">
-        <h1 class="govuk-heading-xl govuk-!-margin-bottom-6">Enter your training provider's <abbr>UK</abbr> Provider Reference Number (<abbr>UKPRN</abbr>)</h1>
+        <noscript>
```
Comment (ChrisJWoodcock) [inline src/SFA.DAS.ProviderRelationships.Web/Views/AccountProviders/Find.cshtml:19]: There is a 'das-js-hide' - can that be used to make seem more balanced?

Comment (cofaulco) [reply src/SFA.DAS.ProviderRelationships.Web/Views/AccountProviders/Find.cshtml]: fair point, done

**Pull request merged**